### PR TITLE
updated to realize search, return results and display detailed weather info 

### PR DESCRIPTION
**Pull request by Pengyong-Zhan**
1. added a section for search and return a result list in index.html
2. added 3 sections to display detailed weather info 
3. updated search.js, weather.js(modules) and weather.js(pages)
4. added an import clause in index.js

### CODE REVIEW

File: src/app/pages/weather.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        liHumidity.append($('<p>').text(humidity + "%"));
+
+        let liWindSpeed = $('<li></li>');
+        liWindSpeed.append($('<p>').text("Wind Speed"));
+        liWindSpeed.append($('<p>').text(windSpeed + "km/h"));
+
+        $('#detailsUl').append(liFeelsLike);
+        $('#detailsUl').append(liVisibility);
+        $('#detailsUl').append(liHumidity);
+        $('#detailsUl').append(liWindSpeed);
```
Comment (hpyhacking) [inline src/app/pages/weather.js:80]: same as above

File: src/app/ui/search.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const Config = require('../modules/config')
+
+const ENDPOINT_GEO = "http://api.openweathermap.org/geo/1.0/direct"
+
+$(document).ready(function() {
+    if ($("meta[name=page]").attr('content') != "index") {
+      return; 
+    }
```
Comment (hpyhacking) [inline src/app/ui/search.js:8]: this should be a page logic, could not shown in a ui module

File: src/index.js [hunk 3]
```diff
@@ -3,3 +3,4 @@ require('./index.scss')
 import "./app/pages/index.js"
 import "./app/pages/weather.js"
 import "./app/pages/contact_us.js"
+import "./app/ui/search.js"
```
Comment (hpyhacking) [inline src/index.js:6]: why add ui module in here?

ui module should be required in a page module, and use a element to call the ui module function.

**Review by hpyhacking — CHANGES_REQUESTED** [review]

File: src/app/modules/weather.js [hunk 4]
```diff
@@ -1,14 +1,31 @@
...
+    this.lon = lon;
+    //this.text = 'Weather Station Example' + Config.API_KEY
   }
 
-  request(lon, lat, callback) {
-    let data = {text: this.text}
-    callback(data)
+  request(callback) {
+    //check cache
+    console.log("test request");
```
[the remark this answers is not in the document]
Comment (Pengyong-Zhan) [reply src/app/modules/weather.js:14]: ok

**Comment by hpyhacking** [comment]
merged in #14

**Pull request closed without merging**